hosted/eks/p1/p1_suite_test.go

```diff
@@ -94,3 +96,81 @@ func updateClusterInUpdatingState(cluster *management.Cluster, client *rancher.C
 	}, "15m", "30s").Should(BeTrue())
 }
+
+func syncK8sVersionUpgradeCheck(cluster *management.Cluster, client *rancher.Client, upgradeNodeGroup bool) {
+	var err error
+	GinkgoLogr.Info("Upgrading cluster to version:" + upgradeToVersion)
+
+	By("upgrading control plane", func() {
+		err = helper.UpgradeEKSClusterOnAWS(region, clusterName, upgradeToVersion)
+		Expect(err).To(BeNil())
+
+		Eventually(func() string {
+			GinkgoLogr.Info("Waiting for k8s upgrade to appear in EKSStatus.UpstreamSpec ...")
+			cluster, err = client.Management.Cluster.ByID(cluster.ID)
+			Expect(err).To(BeNil())
+			return *cluster.EKSStatus.UpstreamSpec.KubernetesVersion
+		}, tools.SetTimeout(5*time.Minute), 10*time.Second).Should(Equal(upgradeToVersion), "Failed while waiting for k8s upgrade to appear in EKSStatus.UpstreamSpec")
+
+		if !helpers.IsImport {
+			// For imported clusters, EKSConfig always has null values; so we check EKSConfig only when testing provisioned clusters
+			for _, ng := range cluster.EKSConfig.NodeGroups {
+				Expect(*ng.Version).To(BeEquivalentTo(k8sVersion), "EKSConfig.NodePools check failed")
+			}
+		}
+	})
+
+	if upgradeNodeGroup {
+		By("upgrading the nodegroup", func() {
+			GinkgoLogr.Info("Upgrading Nodegroup's EKS version")
+			for _, ng := range cluster.EKSStatus.UpstreamSpec.NodeGroups {
+				err = helper.UpgradeEKSNodegroupOnAWS(region, clusterName, *ng.NodegroupName, upgradeToVersion)
+				Expect(err).To(BeNil())
+			}
+
+			Eventually(func() bool {
+				GinkgoLogr.Info("Waiting for the nodegroup upgrade to appear in EKSStatus.UpstreamSpec ...")
+				cluster, err = client.Management.Cluster.ByID(cluster.ID)
+				Expect(err).To(BeNil())
+				for _, ng := range cluster.EKSStatus.UpstreamSpec.NodeGroups {
+					if ng.Version == nil || *ng.Version != upgradeToVersion {
+						return false
+					}
+				}
+				return true
+			}, tools.SetTimeout(5*time.Minute), 10*time.Second).Should(BeTrue(), "Failed while waiting for nodegroup k8s upgrade to appear in EKSStatus.UpstreamSpec")
+
+			if !helpers.IsImport {
+				// For imported clusters, EKSConfig always has null values; so we check EKSConfig only when testing provisioned clusters
+				Expect(*cluster.EKSConfig.KubernetesVersion).To(Equal(upgradeToVersion))
+				for _, ng := range cluster.EKSConfig.NodeGroups {
+					Expect(*ng.Version).To(BeEquivalentTo(upgradeToVersion), "EKSConfig.NodePools upgrade check failed")
+				}
+			}
+		})
+	}
+}
+
+func syncRancherToAWSCheck(cluster *management.Cluster, client *rancher.Client) {
+	var err error
+	loggingTypes := []string{"api", "audit", "authenticator", "controllerManager", "scheduler"}
+	currentNodeGroupNumber := len(cluster.EKSConfig.NodeGroups)
+
+	By("upgrading control plane", func() {
+		syncK8sVersionUpgradeCheck(cluster, client, false)
+	})
+
+	By("adding a NodeGroup", func() {
+		cluster, err = helper.AddNodeGroup(cluster, 1, client, true, true)
+		Expect(err).To(BeNil())
+	})
```

Comment:
nit: Can we also add checks for scaling and deleting the nodegroup while at it?

Reply:
Sure, in my upcoming PR

Reply:
Should we not be checking if these changes are reflected on AWS console as well (not via UpstreamSpec but by fetching the cluster via AWS CLI and testing the changes? Otherwise this test is not very different from our validation tests.

Reply:
I am not sure if we follow the same for other providers sync tests, UpstreamSpec has been source of truth to check cloud changes.
Yes, the tests seems subset of validation test in Sync context.

Reply:
True, that we do not check for other providers since most of the tests have not been automated. 

> Verify the existing details do NOT change in Rancher/AWS console.
Verify the new edits reflect in AWS console.

The test Qase steps mention checking back on AWS console to see the changes have been implemented, and I think that is the point of sync tests, to check if rancher and aws show the same configuration.

Reply:
If UpstreamSpec does not reflect correct values for AWS current state, then its likely a bug.
However I got your point and added validation for TC expected results.